core/src/test/java/org/frankframework/pipes/ReplacerPipeTest.java

```diff
@@ -94,4 +97,106 @@ public void replaceNonXMLCharLongerThanOne() {
 		ConfigurationException e = assertThrows(ConfigurationException.class, this::configurePipe);
 		assertThat(e.getMessage(), Matchers.containsString("replaceNonXmlChar [klkl] has to be one character"));
 	}
+
+	@Test
+	public void testReplaceParameters() throws Exception {
+		pipe.addParameter(ParameterBuilder.create()
+				.withName("varToSubstitute")
+				.withValue("substitutedValue"));
+
+		pipe.addParameter(ParameterBuilder.create()
+				.withName("secondVarToSubstitute")
+				.withValue("secondSubstitutedValue"));
+
+		pipe.setFind("test");
+		pipe.setReplace("head");
+		pipe.configure();
+
+		PipeRunResult res = doPipe(pipe, "<test>?{varToSubstitute} and ?{secondVarToSubstitute}</test>)", session);
+		assertEquals("<head>substitutedValue and secondSubstitutedValue</head>)", res.getResult().asString());
+	}
+
+	@Test
+	public void testSubstituteVarsViaSessionNotSupportedAnymore() throws Exception {
+		session.put("varToSubstitute", "substitutedValue");
+		session.put("secondVarToSubstitute", "secondSubstitutedValue");
+
+		pipe.setFind("test");
+		pipe.setReplace("head");
+		pipe.setSubstituteVars(true);
+		pipe.configure();
+
+		PipeRunResult res = doPipe(pipe, "<test>${varToSubstitute} and ${secondVarToSubstitute}</test>)", session);
+		assertEquals("<head>${varToSubstitute} and ${secondVarToSubstitute}</head>)", res.getResult().asString());
+	}
+
+	@Test
+	public void testSubstituteVars() throws Exception {
+		pipe.setFind("test");
+		pipe.setReplace("head");
+		pipe.setSubstituteVars(true);
+		pipe.configure();
+
+		// application.name is always set in AppConstants.properties
+		PipeRunResult res = doPipe(pipe, "<test>${application.name}</test>)", session);
+		assertEquals("<head>IAF</head>)", res.getResult().asString());
+	}
+
+	@Test
+	@DisplayName("Combine search/replace, parameter substitution and variable resolving")
+	public void combinedSearchAndReplace() throws Exception {
+		pipe.addParameter(ParameterBuilder.create()
+				.withName("parameter1")
+				.withValue("[Parameter value 1]"));
+
+		pipe.addParameter(ParameterBuilder.create()
+				.withName("parameter2")
+				.withValue("[Parameter value 2]"));
+
+		pipe.setFind("test");
+		pipe.setReplace("head");
+		pipe.setSubstituteVars(true);
+		pipe.configure();
+
+		// application.name is always set in AppConstants.properties
+		PipeRunResult res = doPipe(pipe, "<test>?{parameter1} and ${application.name}<br />?{parameter2}</test>", session);
```

Comment:
Je kan het gemakkelijk testen door in src/test/resources/AppConstants.properties het volgende over te nemen:
```
resolve.me=123
unresolved.property=${resolve.me}
```

```suggestion
		PipeRunResult res = doPipe(pipe, "<test>?{parameter1} and ${unresolved.property}<br />?{parameter2}</test>", session);
```

Reply:
Aanpassingen gedaan, dit doet wat ik verwacht. Zou je nog eens kunnen kijken of het nu akkoord is?